app/controllers/posts_controller.rb

```diff
@@ -2,6 +2,25 @@ class PostsController < ApplicationController
   def index
     @users = [User.find(params[:user_id])]
     @posts = @users[0].posts
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      redirect_to user_posts_path(current_user)
+    else
+      render 'new'
+    end
+  end
+
+  def post_params
```

Comment:
### 👋 Hey @Innocent259! 🌟

- [ ] I noticed the routes for creating posts, comments, and likes are missing in your Blog app. 🚧 Let's add them to enable these cool features! 😊

#### example based on your implementation of routes

```
Rails.application.routes.draw do
  root 'users#index'

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:index, :show, :new, :create] # 🗨️ Routes for comments
      resources :likes, only: [:index, :show, :new, :create], controller: 'post_likes' # ❤️‍🔥 Routes for likes
    end
  end
end

```

#### Make sure to implement corresponding controllers to make them work smoothly.